### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException Kotlin @QueryMap Android 6.0.1

### ISSUE THREAD

**Issue by gojkocc** [square/retrofit#2509]
Hi,

I'm using Kotlin to send a QueryMap. With Android newer than 6 everything works without any problem. I guess the problem is the Kotlin generated map that I'm trying to send in combination with Retrofit2. 
For me the Exception doesn't make any sense, the map is a key value pair with the size of 3 to 6. 

Best

`@GET("search?select=(refinements)")
fun getRefinements(@QueryMap refinements: Map<String, String>): LiveData<ApiResponse<SearchRefinementResponse>>`

`Throwing new exception 'length=384; index=1273' with unexpected pending exception: java.lang.ArrayIndexOutOfBoundsException: length=384; index=1273
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at android.arch.lifecycle.LiveData com.x.repositories.my.Repository$CategoryLiveData.createApiCall() (Repository.kt:130)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void com.x.repositories.common.LocalNetworkResourceMediator.fetchFromNetwork(android.arch.lifecycle.LiveData) (LocalNetworkResourceMediator.kt:53)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void com.x.repositories.common.LocalNetworkResourceMediator.access$fetchFromNetwork(com.x.repositories.common.LocalNetworkResourceMediator, android.arch.lifecycle.LiveData) (LocalNetworkResourceMediator.kt:14)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void com.x.repositories.common.LocalNetworkResourceMediator$load$1.onChanged(java.lang.Object) (LocalNetworkResourceMediator.kt:43)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.arch.lifecycle.MediatorLiveData$Source.onChanged(java.lang.Object) (MediatorLiveData.java:119)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.arch.lifecycle.LiveData.considerNotify(android.arch.lifecycle.LiveData$LifecycleBoundObserver) (LiveData.java:129)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.arch.lifecycle.LiveData.dispatchingValue(android.arch.lifecycle.LiveData$LifecycleBoundObserver) (LiveData.java:146)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.arch.lifecycle.LiveData.setValue(java.lang.Object) (LiveData.java:293)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.arch.lifecycle.LiveData$2.run() (LiveData.java:108)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.os.Handler.handleCallback(android.os.Message) (Handler.java:739)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:95)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.os.Looper.loop() (Looper.java:148)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:5417)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:726)
09-19 09:36:34.897 9603-9603/com.x.x A/art: art/runtime/thread.cc:1344] at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:616)`

**Comment by NightlyNexus** [comment]
There's not enough information here to show this is a Retrofit problem. Can you make a test case?
The stacktrace shows something's wrong around `LiveData.setValue`.
It looks like you have a custom CallAdapter installed to create LiveData calls. The error could be there.

**Comment by gojkocc** [comment]
Thanks for your answer.
The exception has to do with Instant Run. When I deactivate Instant Run everything works. https://github.com/square/retrofit/issues/1486

**Comment by sureshjoshi** [comment]
@gojkocc As a small follow-up, was there any correlation with OS version and/or multidex enabled? 

I just ran into this problem (at a demo, of course), but not on my Motorola Lollipop, but it happened on a Motorola Marshmallow or Nougat device (can't recall).

I'd never seen this problem before I enabled multidex, but I don't know if that's a red herring. 

It seems that Instant Run is the underlying culprit, which I need to test - but just wondering if there is anything else at play.
